virtualizarr/tests/test_writers/test_icechunk.py

```diff
@@ -0,0 +1,292 @@
+    from icechunk import IcechunkStore, StorageConfig
+
+    storage = StorageConfig.filesystem(str(tmpdir))
+
+    # TODO if icechunk exposed a synchronous version of .open then we wouldn't need to use asyncio.run here
+    # TODO is this the correct mode to use?
+    store = IcechunkStore.create(storage=storage, mode="w")
+
+    # TODO instead yield store then store.close() ??
+    return store
+
+
+def test_write_new_virtual_variable(
+    icechunk_filestore: "IcechunkStore", vds_with_manifest_arrays: Dataset
+):
+    vds = vds_with_manifest_arrays
+
+    dataset_to_icechunk(vds, icechunk_filestore)
+
+    # check attrs
+    root_group = group(store=icechunk_filestore)
+    assert isinstance(root_group, Group)
+    assert root_group.attrs == {"something": 0}
+
+    # TODO check against vds, then perhaps parametrize?
+
+    # check array exists
+    assert "a" in root_group
+    arr = root_group["a"]
+    assert isinstance(arr, Array)
+
+    # check array metadata
+    # TODO why doesn't a .zarr_format or .version attribute exist on zarr.Array?
+    # assert arr.zarr_format == 3
+    assert arr.shape == (2, 3)
+    assert arr.chunks == (2, 3)
+    assert arr.dtype == np.dtype("<i8")
+    assert arr.order == "C"
+    assert arr.fill_value == 0
+    # TODO check compressor, filters?
+    #
+
+    # check array attrs
+    # TODO somehow this is broken by setting the dimension names???
+    # assert dict(arr.attrs) == {"units": "km"}
+
+    # check dimensions
+    assert arr.attrs["_ARRAY_DIMENSIONS"] == ["x", "y"]
+
+
+def test_set_single_virtual_ref_without_encoding(
+    icechunk_filestore: "IcechunkStore", simple_netcdf4: Path
+):
+    # TODO kerchunk doesn't work with zarr-python v3 yet so we can't use open_virtual_dataset and icechunk together!
+    # vds = open_virtual_dataset(netcdf4_file, indexes={})
+
+    # instead for now just write out byte ranges explicitly
+    manifest = ChunkManifest(
+        {"0.0": {"path": simple_netcdf4, "offset": 6144, "length": 48}}
+    )
+    zarray = ZArray(
+        shape=(3, 4),
+        chunks=(3, 4),
+        dtype=np.dtype("int32"),
+        compressor=None,
+        filters=None,
+        fill_value=None,
+    )
+    ma = ManifestArray(
+        chunkmanifest=manifest,
+        zarray=zarray,
+    )
+    foo = Variable(data=ma, dims=["x", "y"])
+    vds = Dataset(
+        {"foo": foo},
+    )
+
+    dataset_to_icechunk(vds, icechunk_filestore)
+
+    root_group = group(store=icechunk_filestore)
+    array = root_group["foo"]
+
+    # check chunk references
+    # TODO we can't explicitly check that the path/offset/length is correct because icechunk doesn't yet expose any get_virtual_refs method
+
+    expected_ds = open_dataset(simple_netcdf4)
+    expected_array = expected_ds["foo"].to_numpy()
+    npt.assert_equal(array, expected_array)
+
+    ds = open_zarr(store=icechunk_filestore, zarr_format=3, consolidated=False)
+    assert np.allclose(ds.foo.to_numpy(), expected_ds.foo.to_numpy())
```

Comment:
This would test also variable names, dimension names, indexes, and attrs
```suggestion
    import xarray.testing as xrt

    assert xrt.assert_identical(ds, expected_ds)
```